packages/maskbook/src/plugins/ITO/SNSAdaptor/hooks/useClaimablePools.ts

```diff
@@ -24,65 +11,28 @@ export interface SwappedToken {
     unlockTime: Date
 }
 
-export function useClaimablePools(isMainnetOld = false) {
+export function useClaimablePools(chainId: ChainId, isMainnetOld = false) {
     const account = useAccount()
-    const chainId = useChainId()
-    const { ITO_CONTRACT_ADDRESS } = useITOConstants()
-    // Todo: Remove the code after the period that old ITO is being used and continues to be used for a while
-    const { contract: ITO_Contract } = useITO_Contract(isMainnetOld ? ITO_CONTRACT_ADDRESS : undefined)
-
+    const { ITO_CONTRACT_ADDRESS, ITO2_CONTRACT_ADDRESS } = useITOConstants(chainId)
+    const contractAddress = isMainnetOld ? ITO_CONTRACT_ADDRESS : ITO2_CONTRACT_ADDRESS
     //#region fetch claimable pool
-    const { value: poolsFromSubgraph = [], loading: loadingSubgraph } = useClaimablePoolsBySubgraph()
-    const { value: poolsFromWeb3 = [], loading: loadingWeb3 } = useClaimablePoolsByWeb3()
```

Comment:
getPastLogs() through our own rpc is much slower ( > 1 min on mainnet) and unstable (lots of error e.g. CORS on bsc and matic) than thegraph. Remove it temporaily.

Better solution in the future: send a preflight request to thegraph, if it is down, then fallback to `web3.getPastLogs()`